### DISCUSSION TITLE
Multiply items on device production exit recipe

### DISCUSSION THREAD

**Discussion by 776166** [voxeltycoon/issues#913] [Modding]
Now it can be added to config, now it shown in game production interface on device exit but do not working.

**Comment by victorleblais** [comment]
It took me a bit too much to understand the title. It's not _multiply_, it's _multiple_, as in being able to have a recipe that creates more than one item (for example, crushing stone could produce 2 bricks and 1 gravel).

I'd argue that having the possibility to have "drop chances" would be nice. Taking the same example, crushing stone could produce 2 bricks and 20% of the time, produce 1 gravel.

Specifying the desired device output would be great too.

Recipe file could look like this:
```
{
  "TargetUri": "base/crusher.recipetarget",
  "Duration": 5.0,
  "InputItems": [
    { "Uri": "base/stone.item", "Count": 1.0 }
  ],
  "OutputItems": [
    { "Uri": "base/stone_brick.item", "Count": 2.0, "Probability": 1.0, OutputIndex: 0 },
    { "Uri": "base/gravel.item", "Count": 2.0, "Probability": 0.2, OutputIndex: 1 }
  ],
  "ResearchUri": "base/stone_processing.research"
}
```

If you want to go even further, you could imagine having mutually exclusive outputs.
For example, if you crush stone, you have 80% chance of getting 2 bricks **OR** 20% of getting 1 brick and 1 gravel.
The file format could be:
```
  "OutputItems": [
    {
      "Probability": 0.8,
       "OutputItems": [ { "Uri": "base/stone_brick.item", "Count": 2.0, "Probability": 1.0, OutputIndex: 0 } ],
    },
    {
      "Probability": 0.2,
       "OutputItems": [
          { "Uri": "base/stone_brick.item", "Count": 1.0, "Probability": 1.0, OutputIndex: 0 },
          { "Uri": "base/gravel.item", "Count": 1.0, "Probability": 1.0, OutputIndex: 0 }
        ],
    },
  ],
}
```

That would allow A LOT of freedom for mod makers. And the conveyor filter would become extremely useful too.

**Reply by 776166** [reply]
Good Idea. Like Uranium in Factorio. For me it does not needed for now, but it is simply to develop.
